### ISSUE TITLE
Wrong camera is selected on Samsung S9 when cameraPosition: "back" is set

### ISSUE THREAD

**Issue by akandaurov** [opentok/opentok-react-native#349]
# Bug Report

**Current behavior**
Selfie camera is selected on Samsung S9, when I set { cameraPosition: 'back' } in publisher properties.
This is because the device has 3 cameras total (Camera.getNumberOfCameras()) and  the code below
```
            if (cameraPosition.equals("back")) {
                mPublisher.cycleCamera();
            }
```
just switches camera to the 2 instead of 0 (back camera)

**Steps to reproduce**
```
                    <OTPublisher
                        properties={{
                            cameraPosition: 'back',
                        }}
                        style={{ flex: 1 }}
                    />
```
on Samsung S9 and probably some others

**What is the current *bug* behavior?**
Camera 2 is selected (2nd selfie camera)

**What is the expected *correct* behavior?**
Camera 0 is selected (back camera)

**Comment by LuuchoRocha** [comment]
I have the same issue with another samsung phone. Is clear where the issue comes from:

`android/src/main/java/com/opentokreactnative/OTSessionManager.java`

```

    @ReactMethod
    public void initPublisher(String publisherId, ReadableMap properties, Callback callback) {

        String name = properties.getString("name");
        Boolean videoTrack = properties.getBoolean("videoTrack");
        Boolean audioTrack = properties.getBoolean("audioTrack");
        String cameraPosition = properties.getString("cameraPosition");
        Boolean audioFallbackEnabled = properties.getBoolean("audioFallbackEnabled");
        int audioBitrate = properties.getInt("audioBitrate");
        String frameRate = "FPS_" + properties.getInt("frameRate");
        String resolution = properties.getString("resolution");
        Boolean publishAudio = properties.getBoolean("publishAudio");
        Boolean publishVideo = properties.getBoolean("publishVideo");
        String videoSource = properties.getString("videoSource");
        Publisher mPublisher = null;
        if (videoSource.equals("screen")) {
            View view = getCurrentActivity().getWindow().getDecorView().getRootView();
            OTScreenCapturer capturer = new OTScreenCapturer(view);
            mPublisher = new Publisher.Builder(this.getReactApplicationContext())
                    .audioTrack(audioTrack)
                    .videoTrack(videoTrack)
                    .name(name)
                    .audioBitrate(audioBitrate)
                    .resolution(Publisher.CameraCaptureResolution.valueOf(resolution))
                    .frameRate(Publisher.CameraCaptureFrameRate.valueOf(frameRate))
                    .capturer(capturer)
                    .build();
            mPublisher.setPublisherVideoType(PublisherKit.PublisherKitVideoType.PublisherKitVideoTypeScreen);
        } else {
            mPublisher = new Publisher.Builder(this.getReactApplicationContext())
                    .audioTrack(audioTrack)
                    .videoTrack(videoTrack)
                    .name(name)
                    .audioBitrate(audioBitrate)
                    .resolution(Publisher.CameraCaptureResolution.valueOf(resolution))
                    .frameRate(Publisher.CameraCaptureFrameRate.valueOf(frameRate))
                    .build();
            if (cameraPosition.equals("back")) {
                mPublisher.cycleCamera();
            }
        }
        mPublisher.setPublisherListener(this);
        mPublisher.setAudioLevelListener(this);
        mPublisher.setAudioFallbackEnabled(audioFallbackEnabled);
        mPublisher.setPublishVideo(publishVideo);
        mPublisher.setPublishAudio(publishAudio);
        ConcurrentHashMap<String, Publisher> mPublishers = sharedState.getPublishers();
        mPublishers.put(publisherId, mPublisher);
        callback.invoke();
    }

```

As you can see:
```
            if (cameraPosition.equals("back")) {
                mPublisher.cycleCamera();
            }
```
The initPublisher method doesn't accept the `facingMode` parameter and using `cameraPosition` only triggers a `cycleCamera()` and will fail picking the correct camera if the device has more than two cameras.

**Comment by TalhaAhsan** [comment]
This should be fixed now in the latest version of the library. Release notes mention that the cycleCamera() issue with Samsung phones has been fixed in version 2.21.3. You can see [here](https://tokbox.com/developer/sdks/android/release-notes.html).
